### DISCUSSION TITLE
YOLOv5s Bounding box outputs not inside image

### DISCUSSION THREAD

**Discussion by wiky231** [ultralytics/yolov5#7473] [Q&A]
I just trained the YOLOv5s on a custom image dataset on Google Colab. However, after I ran the detection on the test sets, I realised that some of the output bounding boxes are outside of the image. I'm relatively new to this, is there any solution for this?

![JAMSTECTest (3)](https://user-images.githubusercontent.com/103893027/163920724-189737e2-bcf0-409d-a29b-cea00e282a37.jpg)

**Comment by glenn-jocher** [comment]
@axlecky 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. All of the bounding boxes you've shown are contained inside your image.

We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by wiky231** [comment]
Hi thanks for the reply. Sorry what I meant was the confidence scores above the bounding boxes are not within the image. Is there any solution for this?

This is the code I used to perform inference:
`!python detect.py --weights runs/train/exp/weights/best.pt --img 416 --conf 0.1 --source {dataset.location}/test/images`

**Reply by glenn-jocher** [reply]
@axlecky you can customize the Annotator class that plots results if you'd like:
https://github.com/ultralytics/yolov5/blob/ab5b9174940f29a62374bddaf38cd5d2eeb68e25/utils/plots.py#L68-L129

**Reply by wiky231** [reply]
Thanks for the reply! I solved the issue by adding an additional code to change the line thickness as shown below:

`!python detect.py --weights runs/train/exp/weights/best.pt --img 416 --conf 0.1 --source {dataset.location}/test/images --line-thickness 1`

**Reply by wiky231** [reply]
@glenn-jocher, can I also ask if there is any way to use the detect.py code to also generate the mAP scores for the test set? Thanks!

**Reply by glenn-jocher** [reply]
@axlecky detect.py just detects. val.py runs validation against labelled data:
```
python val.py --data coco128.yaml --weights yolov5s.pt
```

**Reply by wiky231** [reply]
thanks @glenn-jocher! Just tried and it worked.

**Reply by wiky231** [reply]
Sorry @glenn-jocher. When I use val.py it does the test on the validation set, is there anyway to change it to the test set? I tried this code but it didnt work.

`!python val.py --task test --weights runs/train/exp/weights/best.pt --data {dataset.location}/data.yaml`

**Reply by glenn-jocher** [reply]
@axlecky yes you can run only any split, i.e. for test split run `python val.py --task test`
https://github.com/ultralytics/yolov5/blob/3f3852e2ff755275098c07fe3bf4d2bde103ab30/val.py#L342

**Reply by wiky231** [reply]
Thanks @glenn-jocher!
